### PR TITLE
Yotint

### PR DESCRIPTION
**Pull request by NikolayOrlov**

### CODE REVIEW

File: projects/vendor/package.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "vendor",
+  "version": "0.0.1",
+  "peerDependencies": {
+    "@angular/common": "^7.2.0",
+    "@angular/core": "^7.2.0",
+    "@spartacus/core": "^0.1.0-prealpha"
```
Comment (LTiger14) [inline projects/vendor/package.json]: We are on alpha at the moment

File: projects/vendor/src/yotpo/baseyotpo/baseyotpo.component.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  getProduct(): Observable<Product> {
+    return this.routingService.getRouterState().pipe(
+      map(state => state.state.params['productCode']),
+      filter(productCode => !!productCode),
+      switchMap((productCode: string) => this.productService.get(productCode))
+    );
+  }
+
+  ngAfterViewInit() {
```
Comment (LTiger14) [inline projects/vendor/src/yotpo/baseyotpo/baseyotpo.component.ts]: This is not called at the moment so you can either remove it or your class should implement the AfterViewInit. That said, I'm not sure what it does.

File: projects/vendor/src/yotpo/baseyotpo/baseyotpo.component.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  constructor(protected elementRef:ElementRef, 
+		protected routingService: RoutingService,
+		protected productService: ProductService) { }
+
+  ngOnInit() {
+    this.product$ = this.getProduct();
+  }
+
+  getProduct(): Observable<Product> {
```
Comment (LTiger14) [inline projects/vendor/src/yotpo/baseyotpo/baseyotpo.component.ts]: Typically component inheritance is not encouraged in angular. You should move the shared logic to a component that is injected in both Yotpo components. Like the `CurrentProductService`.

File: projects/vendor/src/yotpo/yotporeview/yotporeview.component.html [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<ng-container *ngIf="(product$ | async) as product">
```
Comment (LTiger14) [inline projects/vendor/src/yotpo/yotporeview/yotporeview.component.html]: Please run `yarn prettier-fix`

File: projects/vendor/src/yotpo/yotporeview/yotporeview.component.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { Component, ElementRef } from '@angular/core';
+import { BaseyotpoComponent } from './../baseyotpo/baseyotpo.component';
+import { ProductService, RoutingService } from '@spartacus/core';
+
+@Component({
+  selector: 'cx-yotporeview',
+  templateUrl: './yotporeview.component.html',
+  styleUrls: ['./yotporeview.component.css']
```
Comment (LTiger14) [inline projects/vendor/src/yotpo/yotporeview/yotporeview.component.ts]: The files should be `.scss` but for both components we don't have any style you can just delete both css files.

File: projects/vendor/src/yotpo/yotporeview/yotporeview.component.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+})
+export class YotporeviewComponent extends BaseyotpoComponent {
```
Comment (LTiger14) [inline projects/vendor/src/yotpo/yotporeview/yotporeview.component.ts]: Remove the inheritance (see parent class comment)

File: projects/vendor/src/yotpo/yotpostarrating/yotpostarrating.component.html [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<ng-container *ngIf="(product$ | async) as product">
```
Comment (LTiger14) [inline projects/vendor/src/yotpo/yotpostarrating/yotpostarrating.component.html]: Formatting will be fixed with `yarn prettier-fix`

File: projects/vendor/src/yotpo/yotpostarrating/yotpostarrating.component.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import { Component, ElementRef } from '@angular/core';
+import { BaseyotpoComponent } from './../baseyotpo/baseyotpo.component';
+import { ProductService, RoutingService } from '@spartacus/core';
+
+@Component({
+  selector: 'cx-yotpostarrating',
+  templateUrl: './yotpostarrating.component.html',
+  styleUrls: ['./yotpostarrating.component.css']
```
Comment (LTiger14) [inline projects/vendor/src/yotpo/yotpostarrating/yotpostarrating.component.ts]: No need for this stylesheet

File: projects/vendor/src/yotpo/yotpostarrating/yotpostarrating.component.ts [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+})
+export class YotpostarratingComponent extends BaseyotpoComponent {
```
Comment (LTiger14) [inline projects/vendor/src/yotpo/yotpostarrating/yotpostarrating.component.ts]: Also remove inheritance

File: projects/vendor/package.json (hunk 1 above)
Comment (NikolayOrlov) [reply projects/vendor/package.json]: fixed

File: projects/vendor/src/yotpo/baseyotpo/baseyotpo.component.ts (hunk 2 above)
Comment (NikolayOrlov) [reply projects/vendor/src/yotpo/baseyotpo/baseyotpo.component.ts]: fixed

File: projects/vendor/src/yotpo/yotporeview/yotporeview.component.ts (hunk 5 above)
Comment (NikolayOrlov) [reply projects/vendor/src/yotpo/yotporeview/yotporeview.component.ts]: fixed

File: projects/vendor/src/yotpo/yotpostarrating/yotpostarrating.component.ts (hunk 8 above)
Comment (NikolayOrlov) [reply projects/vendor/src/yotpo/yotpostarrating/yotpostarrating.component.ts]: fixed

File: projects/vendor/src/yotpo/yotporeview/yotporeview.component.html (hunk 4 above)
Comment (NikolayOrlov) [reply projects/vendor/src/yotpo/yotporeview/yotporeview.component.html]: applied

File: projects/vendor/src/yotpo/yotpostarrating/yotpostarrating.component.html (hunk 7 above)
Comment (NikolayOrlov) [reply projects/vendor/src/yotpo/yotpostarrating/yotpostarrating.component.html]: applied

File: projects/vendor/src/yotpo/yotpostarrating/yotpostarrating.component.ts (hunk 9 above)
Comment (NikolayOrlov) [reply projects/vendor/src/yotpo/yotpostarrating/yotpostarrating.component.ts]: fixed

File: projects/vendor/src/yotpo/yotporeview/yotporeview.component.ts (hunk 6 above)
Comment (NikolayOrlov) [reply projects/vendor/src/yotpo/yotporeview/yotporeview.component.ts]: fixed

File: projects/vendor/src/yotpo/baseyotpo/baseyotpo.component.ts (hunk 3 above)
Comment (NikolayOrlov) [reply projects/vendor/src/yotpo/baseyotpo/baseyotpo.component.ts]: fixed

File: projects/vendor/src/yotpo/yotporeview/yotporeview.component.ts [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+
+@Component({
+  selector: 'cx-yotporeview',
+  templateUrl: './yotporeview.component.html',
+  styleUrls: []
+})
+export class YotporeviewComponent implements OnInit, AfterViewInit {
+  product$: Observable<Product>;
+  
+  constructor(protected baseyotpoService:BaseyotpoService, protected elementRef: ElementRef) { }
```
Comment (LTiger14) [inline projects/vendor/src/yotpo/yotporeview/yotporeview.component.ts]: This file isn't prettified

File: projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.ts [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+  getProduct(): Observable<Product> {
+    return this.routingService.getRouterState().pipe(
+      map(state => state.state.params['productCode']),
+      filter(productCode => !!productCode),
+      switchMap((productCode: string) => this.productService.get(productCode))
+    );
+  }
+
+  afterViewInit(elementRef: ElementRef) {
+    var s = document.createElement('script');
```
Comment (LTiger14) [inline projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.ts]: Please don't use variables use `const` instead

File: projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.ts [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+    s.type = 'text/javascript';
+    s.text =
```
Comment (LTiger14) [inline projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.ts]: Use a flag to deactivate ts-lint validation for the long string.

File: projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.ts [hunk 13]
```diff
@@ -0,0 +1,37 @@
+import { Injectable, ElementRef } from '@angular/core';
+import { Observable } from 'rxjs';
+import { Product, ProductService, RoutingService, WindowRef } from '@spartacus/core';
+import { filter, map, switchMap } from 'rxjs/operators';
+import { YotpoConfig } from '../yotpoconfig/yotpo-config';
+
+@Injectable({
+ providedIn: 'root',
+})
+export abstract class BaseyotpoService  {
```
Comment (LTiger14) [inline projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.ts]: Could this be just `YotpoService`?

File: tsconfig.json [hunk 14]
```diff
@@ -9,15 +9,27 @@
...
     "paths": {
-      "@spartacus/core": ["projects/core/public_api"],
-      "@spartacus/storefront": ["projects/storefrontlib/src/public_api"]
+      "@spartacus/core": [
+        "projects/core/public_api"
+      ],
+      "@spartacus/storefront": [
+        "projects/storefrontlib/src/public_api"
+      ],
+      "vendor": ["projects/vendor/public_api"]
```
Comment (LTiger14) [inline tsconfig.json]: Please use `@spartacus/vendor` as this is what we are going to release the module as in the future.

File: projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.spec.ts [hunk 15]
```diff
@@ -0,0 +1,12 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { BaseyotpoService } from './baseyotpo.service';
+
+describe('BaseyotpoService', () => {
```
Comment (LTiger14) [inline projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.spec.ts]: Please provide testing for the functions of the service.

File: projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.ts [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+
+  getProduct(): Observable<Product> {
+    return this.routingService.getRouterState().pipe(
+      map(state => state.state.params['productCode']),
+      filter(productCode => !!productCode),
+      switchMap((productCode: string) => this.productService.get(productCode))
+    );
+  }
+
+  afterViewInit(elementRef: ElementRef) {
```
Comment (LTiger14) [inline projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.ts]: Could you give this function a name that provides a better description of the usage.

File: projects/vendor/src/yotpo/yotporeview/yotporeview.component.ts [hunk 17]
```diff
@@ -0,0 +1,24 @@
+import { Component, OnInit, AfterViewInit, ElementRef } from '@angular/core';
+import { BaseyotpoService } from './../baseyotpo/baseyotpo.service';
+import { Observable } from 'rxjs';
+import { Product } from '@spartacus/core';
+
+@Component({
+  selector: 'cx-yotporeview',
+  templateUrl: './yotporeview.component.html',
```
Comment (LTiger14) [inline projects/vendor/src/yotpo/yotporeview/yotporeview.component.ts]: Typically for components and files the naming would be something more along the lines of `yotpo-review.component...` and the selector would be `cx-yotpo-review`.

File: projects/vendor/README.md [hunk 18]
```diff
@@ -0,0 +1,24 @@
+# Vendor
```
Comment (LTiger14) [inline projects/vendor/README.md]: Could you update this with a quick descriptions of the steps to use this. No need to be super polished our writer will edit it as long as there is the basic.

File: projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.ts [hunk 19]
```diff
@@ -0,0 +1,37 @@
+import { Injectable, ElementRef } from '@angular/core';
```
Comment (LTiger14) [inline projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.ts]: Please name the parent folder service so we can add more services to it eventually.

File: projects/vendor/public_api.ts [hunk 20]
```diff
@@ -0,0 +1,8 @@
+/*
+ * Public API Surface of vendor
+ */
+
+export * from './src/yotpo/yotpo.module';
```
Comment (LTiger14) [inline projects/vendor/public_api.ts]: In order to make this cleaner. For the yotpo folder you should have create an `index.ts` that exports everything that has to be exported by yotpo and export this index file in the public_api.

File: projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.ts (hunk 11 above)
Comment (NikolayOrlov) [reply projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.ts]: fixed. thanks

File: projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.ts (hunk 12 above)
Comment (NikolayOrlov) [reply projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.ts]: fixed. thanks

File: projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.ts (hunk 13 above)
Comment (NikolayOrlov) [reply projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.ts]: fixed

File: projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.ts (hunk 16 above)
Comment (NikolayOrlov) [reply projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.ts]: done

File: projects/vendor/src/yotpo/yotporeview/yotporeview.component.ts (hunk 17 above)
Comment (NikolayOrlov) [reply projects/vendor/src/yotpo/yotporeview/yotporeview.component.ts]: done. thanks

File: projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.ts (hunk 19 above)
Comment (NikolayOrlov) [reply projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.ts]: done

File: tsconfig.json (hunk 14 above)
Comment (NikolayOrlov) [reply tsconfig.json]: done

File: projects/vendor/src/yotpo/yotporeview/yotporeview.component.ts (hunk 10 above)
Comment (NikolayOrlov) [reply projects/vendor/src/yotpo/yotporeview/yotporeview.component.ts]: fixed. thanks

File: projects/vendor/public_api.ts (hunk 20 above)
Comment (NikolayOrlov) [reply projects/vendor/public_api.ts]: done

File: projects/vendor/README.md (hunk 18 above)
Comment (NikolayOrlov) [reply projects/vendor/README.md]: done basically

File: projects/vendor/src/yotpo/service/yotpo.service.ts [hunk 21]
```diff
@@ -0,0 +1,49 @@
...
+      map(state => state.state.params['productCode']),
+      filter(productCode => !!productCode),
+      switchMap((productCode: string) => this.productService.get(productCode))
+    );
+  }
+
+  addYotpoInitWidgetsScript(elementRef: ElementRef) {
+    const s = document.createElement('script');
+    s.type = 'text/javascript';
+    // tslint:disable-next-line
```
Comment (LTiger14) [inline projects/vendor/src/yotpo/service/yotpo.service.ts]: This doesn't work as it's not the line under. You should do a /* tslint:disable */ line 44 and /* tslint:enable */ line 47

Comment (NikolayOrlov) [reply projects/vendor/src/yotpo/service/yotpo.service.ts]: corrected

File: projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.spec.ts (hunk 15 above)
Comment (NikolayOrlov) [reply projects/vendor/src/yotpo/baseyotpo/baseyotpo.service.spec.ts]: done

**Comment by giancorderoortiz** [comment]
Closing due to many months of inactivity.
If still relevant, please update PR with latest develop branch, and re-open with meaningful comments so that it can be evaluated by development team.

**Pull request closed without merging**